src/sentry/rules/conditions/event_frequency.py

```diff
@@ -110,10 +113,11 @@ def passes(self, event, state):
         if not interval:
             return False
 
-        current_value = self.get_rate(event, interval, self.rule.environment_id)
+        # TODO(mgaeta): Bug: Rule is optional.
```

Comment:
Yeah I'd like to do that. Currently it's only being types as optional because of the constructor:
```py
class RuleBase(metaclass=RuleDescriptor):
    ...
    def __init__(self, project, data=None, rule=None):
        ...
        self.rule = rule
```